### PR TITLE
feat(storage): support max version pinning duration

### PR DESCRIPTION
**Pull request by zwang28**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

This PR adds an upper limiter for hummock version pinning duration, to avoid blocking storage GC infinitely.

A long hummock pinning duration can be caused by
- slow hummock iteration.
- bug that doesn't unpin version as expected.

The new config `max_version_pinning_duration_sec` is 3 hours by default. A version pinned by compute node will be unpinned after `max_version_pinning_duration_sec`. After that, there is no guarantee of success for hummock reading upon this version, because SSTs referenced by it may have been GCed.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by Li0k — APPROVED** [review]
LGTM, thanks for the PR

**Review by hzxa21 — APPROVED** [review]
Rest LGTM.

**Comment by hzxa21** [comment]
Not related to this PR. Do you think we need to add a timeout for RowSeqExecutor and storage call including `get` and `iter` as well?

File: src/storage/src/hummock/local_version/pinned_version.rs [hunk 1]
```diff
@@ -180,21 +181,34 @@ pub(crate) async fn start_pinned_version_worker(
...
 
-    let mut version_ids_in_use: BTreeMap<u64, usize> = BTreeMap::new();
-
+    let mut version_ids_in_use: BTreeMap<u64, (usize, Instant)> = BTreeMap::new();
+    let max_version_pinning_duration_sec = Duration::from_secs(max_version_pinning_duration_sec);
     // For each run in the loop, accumulate versions to unpin and call unpin RPC once.
     loop {
         min_execute_interval_tick.tick().await;
+        // 0. Expire versions.
+        while version_ids_in_use.len() > 1 && let Some(e) = version_ids_in_use.first_entry() {
```
[the remark this answers is not in the document]
Comment (zwang28) [reply src/storage/src/hummock/local_version/pinned_version.rs:190]: To not break the fact that there is always at least one item in `version_ids_in_use`.

**Pull request merged**